### PR TITLE
Fix for icon validation bug

### LINKED ISSUE #12653: Icon validation bug
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.4.3(10.0.1 also checked)

### Bug summary

You can submit an item without chosen icon (where it is necessary parameter) . There is even no warnings before you will try to save content.
image

### Specifics

_No response_

### Steps to reproduce

Make blocklist with items which should contain an image as necessary parameter  

### Expected result / actual result

Expected to have an error on submitting , or warning before you choose icon

### PR DESCRIPTION
**Pull request by ustadstar**
Fix required image validation when saving in a blocklist

### Prerequisites

- [ ] I have added steps to test this contribution in the description 

If there's an existing issue for this PR then this fixes #12653

### Description

The media picker required validation was not working client side.

Required was set like this:

`<input type="hidden" name="modelValue" ng-model="vm.model.value.length" ng-required="vm.model.validation.mandatory && vm.model.value.length === 0" />`

this checks if the mandatory option is true for the property and if none has already been selected. The issue with this solution is that it does not update or check validation. I have copied a validateMandatory function from colorpicker. Now the validation works inside a block list editor.

### How to test
Make an block list item, which contains a required image field. Like in the screenshot. Leave the field empty, click create, you should see an error.

image

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker3/umbMediaPicker3PropertyEditor.component.js [hunk 1]
```diff
@@ -396,6 +397,19 @@
             });
         }
 
+        function validateMandatory() {
+            debugger;
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker3/umbMediaPicker3PropertyEditor.component.js:401]: @ustadstar Make sure to remove the debugger again! 😄

Comment (ustadstar) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker3/umbMediaPicker3PropertyEditor.component.js]: @nul800sebastiaan my bad.. I have removed it!

**Comment by ustadstar** [comment]
@nul800sebastiaan I have resolved all issues, is this PR ready to merge?

**Comment by nul800sebastiaan** [comment]
Very sorry for the delay on this one @ustadstar! I've finally had a look at this and unfortunately it's not just always blocking the image picker, even if you pick something! Could you have another look and see if you can reset the picker once an item has been picked?

Here's what it looks like at the moment: 


https://user-images.githubusercontent.com/304656/227986555-7dfe3972-09cf-490a-b228-113f40cd2b34.mp4

**Comment by nul800sebastiaan** [comment]
Hey there @ustadstar! It's been a while since we've heard from you. We'd love to keep going on this one but it seems like it doesn't fit in for you at the moment. In order to keep things tidy, we'll close this one for now. Please let us know when you can get back to it, we can either resurrect this PR or go for an attempt number 2! Thanks again for all your work on this and we hope to hear from you soon! 🙌

**Pull request closed without merging**